src/components/CollaboratorModal/submodals/MainSubmodal.tsx

```diff
@@ -119,20 +141,17 @@ const MainSubmodal = () => {
           isDisabled={collaboratorRoleData?.role !== "ADMIN"}
         >
           <FormLabel>Only admins can add or remove collaborators</FormLabel>
-          <Input
-            onChange={(event) => {
-              setNewCollaboratorEmail(event.target.value)
-              setAddCollaboratorError("")
-            }}
-          />
-          <FormErrorMessage>{addCollaboratorError}</FormErrorMessage>
+          <Input {...register("newCollaboratorEmail")} />
+          <FormErrorMessage>{errorMessage}</FormErrorMessage>
```

Comment:
Are we able to reset the form state when the modal is closed, and after any change? Here's the steps to reproduce:
1. Open the collaborators modal (main modal)
2. Enter an invalid email address into the input box
3. Click on "Add collaborator", the error message will prompt the user that the email is invalid.
4. Change the input to a valid email address
5. The "Add collaborator" button is still disabled and the error message is still shown. (expected: error message disappears on change in the email, and the button is re-enabled)

The same applies even when the modal is closed and re-opened. The difference is the input box is cleared, but the error message still shows and the "Add collaborators" button is still disabled.

Reply:
this is a good point and i didn't have any good solution to this; i need to check with @NatMaeTan on this flow as i'm of the opinion that we should validate for email here and check for existence later (and feedback using a toast)

Reply:
Hmm the point is a bit separate from the one below. This one applies generally to any errors that the user encounters, not just having an invalid email address. As long as the user encounters an error, even if the user corrects the input to something valid, they are not able to add collaborator unless they refresh the page (the button is always disabled and the error message persists).